spec/features/reservations_spec.rb

```diff
@@ -545,4 +545,24 @@
       end
     end
   end
+  context 'redirects properly' do
+    before(:each) { empty_cart }
+    it 'handles direct url visit to /reservation/new with empty cart' do
+      visit root_path
+      visit new_reservation_path
+      expect(page.current_url).to include(root_path)
+      expect(page).to have_content('Catalog')
+    end
+    it 'handles press reserve button with empty cart' do
+      sign_in_as_user(@admin)
+      visit root_path
+      first('.equipment_title_link').click
```

Comment:
We don't need to have all of these clicks and expectations in this test. Fundamentally, we're testing what happens when you either visit `/reservations/new` directly or click the 'Reserve' button with an empty cart, so we should only include actions necessary to test those conditions (i.e. not including tests for an intermediate page visit).